hbase-server/src/main/java/org/apache/hadoop/hbase/master/balancer/SimpleLoadBalancer.java

```diff
@@ -594,10 +597,13 @@ private void addRegionPlan(final MinMaxPriorityQueue<RegionPlan> regionsToMove,
     regionsToReturn.add(rp);
   }
 
+  /**
+   * Override to invoke {@link #setClusterLoad} before balance
+   */
   @Override
-  public List<RegionPlan> balanceCluster(TableName tableName,
-      Map<ServerName, List<RegionInfo>> clusterState) throws HBaseIOException {
-    LOG.debug("Start Generate Balance plan for table: " + tableName);
-    return balanceCluster(clusterState);
+  public synchronized List<RegionPlan>
+      balanceCluster(Map<TableName, Map<ServerName, List<RegionInfo>>> clusterLoadPerTable) {
+    setClusterLoad(clusterLoadPerTable);
```

Comment:
Add reason for why need to setClusterLoad?

Reply:
ok